observability/micrometer/src/main/java/org/wildfly/extension/micrometer/registry/ApplicationRegistry.java

```diff
@@ -0,0 +1,126 @@
+package org.wildfly.extension.micrometer.registry;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.search.RequiredSearch;
+import io.micrometer.core.instrument.search.Search;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.ToDoubleFunction;
+import java.util.stream.StreamSupport;
+
+public class ApplicationRegistry extends SimpleMeterRegistry {
+    private static final String TAG_WF_DEPLOYMENT = "WF_DEPLOYMENT";
+    private final String deploymentName;
+    private final MeterRegistry parentRegistry;
+
+    public ApplicationRegistry(String deploymentName, MeterRegistry parentRegistry) {
+        this.deploymentName = deploymentName;
+        this.parentRegistry = parentRegistry;
+    }
+
+    @Override
+    public List<Meter> getMeters() {
+        return parentRegistry.getMeters().stream().filter(m -> !containsWrongDeploymentTag(m.getId()))
+                .toList();
+    }
+
+    @Override
+    public void forEachMeter(Consumer<? super Meter> consumer) {
+        parentRegistry.forEachMeter(m -> {
+            if (!containsWrongDeploymentTag(m.getId())) {
+                consumer.accept(m);
+            }
+        });
+    }
+
+    @Override
+    public Search find(String name) {
+        // ???
+        return parentRegistry.find(name).tag(TAG_WF_DEPLOYMENT, deploymentName);
```

Comment:
Note to self and others....

AIUI

1) The tag added here cannot be removed by the caller.
2) The various Search methods that return meters from the Search all use a logical AND. So adding another tag with the same name but a different value doesn't 'override' what is done here.

So, it looks like what's done here prevents these Search objects being used as a way to extract meters that are would not be accessible via the other MeterRegistry methods.

I haven't looked much at ApplicationRegistryTest but ^^^ would be good things to cover in that kind of unit test.